### PR TITLE
Spacetime/Override Deletion

### PR DESCRIPTION
**Pull request by edwardneo**
- Added button and popup for coordinators to delete a spacetime for a section
- Added button and popup for coordinators and mentors to delete an override for a section

- Created a new modal to verify deletion
- Added two views to spacetime.py to handle functionality
- Comprehensive tests (requiring a Coordinator Factory to be made)

### CODE REVIEW

File: csm_web/frontend/src/components/section/SpacetimeEditModal.tsx [hunk 1]
```diff
@@ -36,6 +38,7 @@ export default class SpacetimeEditModal extends React.Component<SpacetimeEditMod
       time: timeString.slice(0, sliceIndex),
       isPermanent: false,
       changeDate: "",
+      override: props.overrideEdit,
```
Comment (smartspot2) [inline csm_web/frontend/src/components/section/SpacetimeEditModal.tsx:41]: I don't believe there is any need to make this a state; this variable won't be changing in the lifecycle of the component, and is always going to be the same as the prop.

File: csm_web/frontend/src/components/section/SpacetimeEditModal.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ interface SpacetimeEditModalProps {
   closeModal: () => void;
   defaultSpacetime: Spacetime;
   reloadSection: () => void;
+  overrideEdit: boolean;
```
Comment (smartspot2) [inline csm_web/frontend/src/components/section/SpacetimeEditModal.tsx:13]: [nit] imo, `isEditingOverride` or `editingOverride` seems more intuitively a boolean; this is incredibly tiny, so feel free to ignore if you feel differently.

**Review by smartspot2 — CHANGES_REQUESTED** [review]
Overall, this looks really good; most of my comments are for smaller things/stylistic things, but I'll summarize a few points here.

For the frontend, there's mostly just naming convention nitpicks, and some styling changes (ex. spacing, alignment); beyond that, the functionality for the UI looks great.

For the backend, the main things I commented on are to add logging and error messages, which are also just small nitpicks. Additionally (and more importantly), I don't see any tests currently, and I'd strongly suggest adding tests before deploying, to ensure that all edge cases are handled correctly (especially with timing in deleting spacetimes and overrides).

Lastly, I'd recommend changing this PR to a non-draft PR when you push your next changes, and I'd also recommend changing the title and description of the PR to be something more descriptive for this project. That is, add a few sentences/bullet points describing exactly what this project is, and an overall list of things changed through this pull request.

Keep up the good work!

File: csm_web/frontend/src/components/section/SpacetimeDelete.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  }
+
+  function deleteType() {
+    if (overrideDelete) {
+      return "Override";
+    } else {
+      return "Spacetime";
+    }
+  }
```
[the remark this answers is not in the document]
Comment (noortor) [reply csm_web/frontend/src/components/section/SpacetimeDelete.tsx:35]: replace function calls with ternary operators `{overrideDelete ? "Override": "Spacetime"}`

**Review by noortor — CHANGES_REQUESTED** [review]

**Comment by smartspot2** [comment]
@kennethsieu @edwardneo Is there any update on this PR? The last changes were from 8 days ago.

**Review by noortor — APPROVED** [review]

**Review by smartspot2 — APPROVED** [review]
